mycore-base/src/main/java/org/mycore/frontend/cli/MCRObjectCommands.java

```diff
@@ -702,36 +771,34 @@ private static Transformer getTransformer(String style) {
     }
 
     /**
-     * The method read a MCRObject and use the transformer to write the data to a file. They are any steps to handel
-     * errors and save the damaged data.
+     * The method read a MCRObject and use a transformation to write the data to a file. There aren't any steps to
+     * handel errors and save the damaged data.
      * <ul>
-     * <li>Read data for object ID in the MCRObject, add ACL's and store it as checked and transformed XML. Return true.
+     * <li>Read data for object ID in the MCRObject, add ACLs and store it as checked and transformed XML. Return true.
      * </li>
-     * <li>If it can't find a transformer instance (no script file found) it store the checked data with ACL's native in
+     * <li>If it can't find a transformer instance (no script file found) it store the checked data with ACLs native in
      * the file. Warning and return true.</li>
-     * <li>If it get an exception while build the MCRObject, it try to read the XML blob and stor it without check and
-     * ACL's to the file. Warning and return true.</li>
-     * <li>If it get an exception while store the native data without check, ACÖ's and transformation it return a
+     * <li>If it get an exception while build the MCRObject, it try to read the XML blob and store it without check and
+     * ACLs to the file. Warning and return true.</li>
+     * <li>If it get an exception while store the native data without check, ACLs and transformation it return a
      * warning and false.</li>
      * </ul>
      *
      * @param dir
      *            the file instance to store
      * @param trans
-     *            the XML transformer
+     *            the transformation
      * @param nid
      *            the MCRObjectID
      * @return true if the store was okay (see description), else return false
-     * @throws TransformerException
      * @throws IOException
      * @throws MCRException
-     * @throws SAXParseException
      */
-    private static boolean exportMCRObject(File dir, Transformer trans, String nid)
-        throws TransformerException, IOException, MCRException, SAXParseException {
+    private static boolean exportMCRObject(File dir, BiConsumer<MCRContent, OutputStream> trans, String nid) throws
```

Comment:
All implementations throw some kind of error, maybe it would be better to write a own specific interface for that, or we could implement or reuse a ThrowingBiConsumer. We could use https://github.com/pivovarit/throwing-function or could implement a own Interface like https://github.com/mycore-org/mycore/blob/6795c535d118e7b541a241497b7514404449dd2c/mycore-base/src/main/java/org/mycore/common/function/MCRThrowFunction.java .

Reply:
I've considered that myself and even implemented a `MCRThrowBiConsumer<T,U,E>` that used the same design principles as the other `MCRThrow`-classes.

The problem is line 628 where the stylesheet implementation throws multiple distinct exceptions:
```
} catch (TransformerException | SAXParseException e) {
```

The two exception types have no common parent other than `Exception`, so we would either have to settle for
`MCRThrowBiConsumer<MCRContent, OutputStream, Exception>`, by which we would gain nothing, or we would have to introduce `MCRBiThrowBiConsumer<T,U,E1,E2>`, which is clearly ridicules.

Reply:
But I'm open for other suggestions,

Reply:
I would vote against using the pivovarit functions to avoid another dependency to another project and implement the interfaces ourselves if necessary.

Reply:
I agree. we should stick to `MCRThrow`-... instead of including another dependency.

Additionaly, [`ThrowingBiConsumer`](https://github.com/pivovarit/throwing-function/blob/master/src/main/java/com/pivovarit/function/ThrowingBiConsumer.java) also supports only one exception type.